content/working_with/official_plugins/Infrastructure/starlingx.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: bt_wiki
+title: StarlingX Plugin
+category: Official Plugins
+draft: false
+weight: 100
+aliases:
+    - /plugins/starlingx/
+    - /developer/official_plugins/starlingx/
+---
+
+The StarlingX plugin enables you to discover StarlingX systems and manage resources on StarlingX systems.
+
+Start out by providing the credentials for an existing StarlingX system. The install workflow will gather information from the system, for example type and location.
+
+To discover subclouds, execute the `discover_and_deploy` workflow. This workflow performs the following tasks:
+  - Discover all subclouds that are managed by the system controller (parent).
+  - Create a deployment group.
+  - Create a deployment for each discovered subcloud. Add the subcloud deployment to the deployment group.
+  - Execute batch install on the deployment group.
+
+
+__NOTE: The StarlingX Plugin is designed to work with the StarlingX blueprint, which is located in the examples directory of the plugin repository.__
+
+
+# Requirements
+
+* Python versions:
+  * 3.6.x
+
+# Labels
+
+The following are possible for a StarlingX system deployment:
+
+  - `csys-obj-type`
+  - `csys-env-type`
+  - `csys-obj-parent`
+  - `csys-location-name`
+  - `csys-location-lat`
+  - `csys-location-long`
+  - `csys-wrcp-services`
+
+# Node Types
+
+## **cloudify.nodes.starlingx.WRCP**
+
+This node represents a StarlingX System. A system can be a System Controller, Standalone system, or a Subcloud.
+
+**Properties**
+
+  * `use_external_resource`: Always true. This parameters indicates that the resource already exists.
+
+  * `client_config`: A dictionary containing the following keys:
+    * `auth_url`: The StarlingX system endpoint, including the protocol, IP, port, and path, e.g. `https://172.20.0.1:5000/v3`. IPv6 systems' auth urls, should be provided with the IP surrounded by square brackets, e.g. `https://[2001:0db8:85a3:0000:0000:8a2e:0370:7334]:5000/v3`.
+    * `username`: The StarlingX username.
+    * `api_key`: The StarlingX password.
+    * `project_name`: The client project name, default: `admin`.
+    * `user_domain_name`: The client user domain name, default: `Default`.
+    * `project_domain_name`: The client project domain name, default: `Default`.
+    * `region_name`: The region name, either `RegionOne` for a system controller, or the subcloud name in the case that the system is a subcloud.
+    * `insecure`: Whether to ignore certification validation when using `https` as the protocol..
+    * `cacert`: The content of a Certificate Authority when using `https` as the protocol.
+
+  * `resource_config`: Parameters that describe the system in the StarlingX API. Currently, there is no need to provide these parameters.
+
+**Runtime Properties**:
```

Comment:
The flow also sets a site per Subcloud and adds it if it does not exist. not sure in which section but it is worth mentioning it here.